tests/functional/ovndbcluster_controller_test.go

```diff
@@ -27,11 +28,152 @@ import (
+						clusterName = OVNSBDBClusterName
+					}
+					statefulSetName := types.NamespacedName{
+						Namespace: cluster.Namespace,
+						Name:      "ovsdbserver-" + DnsName,
+					}
+
+					Eventually(func(g Gomega) int {
+						listDns := GetDNSDataList(statefulSetName)
+						return len(listDns.Items)
+					}).Should(BeNumerically("==", 3))
+
+					// Scale down to 1
+					Eventually(func(g Gomega) {
+						c := GetOVNDBCluster(clusterName)
+						*c.Spec.Replicas = 1
+						g.Expect(k8sClient.Update(ctx, c)).Should(Succeed())
+					}).Should(Succeed())
+
+					// Check if dnsdata CR is down to 1
+					Eventually(func() int {
+						targetedDnsOcurrences := 0
+						listDns := GetDNSDataList(statefulSetName)
+						// This calls CreateOVNDBClusters at the BeforeEach, which will
+						// create NB and SB cluster, both belonging at the same namespace
+						// GetDNSDataList will return ALL occurrences of DNSData (will include
+						// the NB and the SB occurences). Since both clusters were created with 3
+						// replicas and only the targeted one has been scaled down to 1 if we don't
+						// filter by name the result would be 4 (3+1)
+						for _, d := range listDns.Items {
+							if strings.Contains(d.Name, DnsName) {
+								targetedDnsOcurrences++
+							}
+						}
+						return targetedDnsOcurrences
+					}).Should(BeNumerically("==", 1))
+				},
+				Entry("DNS entry NB", "sb"),
+				Entry("DNS entry SB", "nb"),
+			)
+			DescribeTable("Should update DNSData IP if pod IP changes",
+				func(DNSEntryName string) {
+					var clusterName types.NamespacedName
+					var OVNSBDBClusterName types.NamespacedName
+					var cluster *ovnv1.OVNDBCluster
+					var dbs []types.NamespacedName
+
+					_ = th.CreateNetworkAttachmentDefinition(types.NamespacedName{Namespace: namespace, Name: "internalapi"})
+					dbs = CreateOVNDBClusters(namespace, map[string][]string{namespace + "/internalapi": {"10.0.0.1"}}, 3)
+					OVNSBDBClusterName = types.NamespacedName{Name: dbs[1].Name, Namespace: dbs[1].Namespace}
+					cluster = GetOVNDBCluster(OVNSBDBClusterName)
+					clusterName = OVNSBDBClusterName
+					var FullDNSEntryName = DNSEntryName + "." + cluster.Namespace + ".svc"
+
+					// Check that DNSData info has been created with correct IP (10.0.0.1)
+					Eventually(func(g Gomega) {
+						// Get DNSData
+						dnsEntry := GetDNSData(types.NamespacedName{Name: "dns-ovsdbserver-sb-0", Namespace: cluster.Namespace})
+						// DNSData contains Hosts (an array), this array will contain
+						// ovsdbserver-sb-0.ns.svc and ovsdbserver-sb.ns.svc, need to get
+						// in which index is located the one which the flavor is currently testing (DNS Cname or DNS)
+						index := -1
+						for i, element := range dnsEntry.Spec.Hosts {
+							for _, el_host := range element.Hostnames {
+								if el_host == FullDNSEntryName {
+									index = i
+								}
+							}
+						}
+						g.Expect(index).To(BeNumerically(">=", 0))
```

Comment:
nit: what you actually care about is that the name was found, not which index it was. So you could instead have a `found` boolean and set it in line 119+; then check here that it's True.

This would better reflect the intent of the check.

Reply:
you're correct, though index is also needed, as I need to check the test-flavor DNS entry ("CName" vs A record), I was using index as a boolean, but I can explicitly add the boolean to make the test clearer :).
Done